test/e2e/listorgetkubernetesobjects.go

```diff
@@ -0,0 +1,116 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/redhatopenshift"
+)
+
+var _ = Describe("ListOrGetKubernetes Objects [ListOrGetKubernetes]", func() {
```

Comment:
This will not be runnable against prod, and will need special configuration to run in int (will need access to an admin client cert + configuring a client suitably).

Please ensure this works in int and that it is easily disable-able in prod.

Most of this feels like it could/should be handled in unit test - unit test all the cases, then do one e2e that can imply that the unit test validity carries forward to real life.